cmd/storage-class_test.go

```diff
@@ -0,0 +1,280 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	"reflect"
+	"testing"
+)
+
+func TestParseStorageClass(t *testing.T) {
+	ExecObjectLayerTest(t, testParseStorageClass)
+}
+
+func testParseStorageClass(obj ObjectLayer, instanceType string, t TestErrHandler) {
+	tests := []struct {
+		name            int
+		storageClassEnv string
+		wantSc          storageClass
+		expectedError   error
+	}{
+		{1, "EC:3", storageClass{
+			Scheme: "EC",
+			Parity: 3},
+			nil},
+		{2, "EC:4", storageClass{
+			Scheme: "EC",
+			Parity: 4},
+			nil},
+		{3, "AB:4", storageClass{
+			Scheme: "EC",
+			Parity: 4},
+			errors.New("Unsupported scheme AB. Supported scheme is EC")},
+		{4, "AB:4:5", storageClass{
```

Comment:
it is less confusing for the reader if we have exactly one thing wrong about the input being parsed. E.g "AB:4:5" has both scheme and number of sections wrong. Since scheme related failure test case is already present, we could use "EC:4:5" as the invalid input to this unit test.